### PR TITLE
feat: generic table components

### PR DESCRIPTION
**Pull request by ArkuVonSymfon**
https://github.com/Layer-Fi/layer-react/assets/32065451/906d380e-9c95-480b-92e4-42efd1d6cf8c

### CODE REVIEW

File: src/components/TableBody/TableBody.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+import { TableHeadProps } from '../../types/table'
+import classNames from 'classnames'
+
+export const TableBody = ({ children }: TableHeadProps) => {
```
Comment (tomaszantas) [inline src/components/TableBody/TableBody.tsx:5]:
```suggestion
export const TableBody = ({ children }: TableBodyProps) => {
```

File: src/components/TableBody/TableBody.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  const tbodyClassNames = classNames('Layer__table-body')
```
Comment (tomaszantas) [inline src/components/TableBody/TableBody.tsx:6]: single class, so I believe that `classNames` is no needed here

File: src/components/TableHead/TableHead.tsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+import { TableHeadProps } from '../../types/table'
+import classNames from 'classnames'
+
+export const TableHead = ({ children }: TableHeadProps) => {
+  const theadClassNames = classNames('Layer__table-header')
+
+  return <thead className={theadClassNames}>{children}</thead>
+}
```
Comment (tomaszantas) [inline src/components/TableHead/TableHead.tsx:9]:
```suggestion
export const TableHead = ({ children }: TableHeadProps) => (
  <thead className='Layer__table-header'>{children}</thead>
)
```

File: src/icons/ChevronDownFill.tsx [hunk 4]
```diff
@@ -4,7 +4,7 @@ import { IconSvgProps } from './types'
 const ChevronDownFill = ({ size = 18, ...props }: IconSvgProps) => (
   <svg
     xmlns='http://www.w3.org/2000/svg'
-    viewBox='0 0 18 18'
+    viewBox={`0 0 18 18`}
```
Comment (tomaszantas) [inline src/icons/ChevronDownFill.tsx:7]: do we need this change here?

File: src/types/table.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+export interface TableContextProps {
+  expandedRows: string[]
+  setExpandedRows: (rowKey: string) => void
+  collapseRows: () => void
+}
+
+export interface TableProps {
+  children: React.ReactNode | React.ReactNode[]
```
Comment (tomaszantas) [inline src/types/table.ts:8]: Let's try to deconstruct these elements, for example:

```
import { ReactNode } from 'react'

(...)
children: ReactNode
(...)
```

This probably should also work with `Fragment`

**Review by tomaszantas — CHANGES_REQUESTED** [review]

**Review by oliwia1003 — APPROVED** [review]
LGTM

**Review by tomaszantas — APPROVED** [review]
LGTM!

**Pull request merged**